site/app/controllers/forum/ForumController.php

```diff
@@ -624,8 +673,13 @@ public function getSinglePost() {
     #[Route("/courses/{_semester}/{_course}/forum/announcements", methods: ["POST"])]
     public function alterAnnouncement(bool $type) {
         $thread_id = $_POST["thread_id"];
+        if ($type) {
+            $this->sendSocketMessage(['type' => 'announce_thread', 'thread_id' => $thread_id]);
+        }
+        else {
+            $this->sendSocketMessage(['type' => 'unpin_thread', 'thread_id' => $thread_id]);
+        }
```

Comment:
This would be a great place to use a ternary operator.
```suggestion
        $this->sendSocketMessage([
            'type' => $type ? 'announce_thread' : 'unpin_thread',
            'thread_id' => $thread_id,
        ]);
```